### DISCUSSION TITLE
Try to use Shiny.BluetoothLE library for Avalonia

### DISCUSSION THREAD

**Discussion by redradist** [shinyorg/shiny#1519] [Q&A]
Is there some example showing how to use Shiny.BluetoothLE in Avalonia Android and iOS application ??
Cannot make it to work ...

**Comment by aritchie** [comment]
I don't use avalonia at all, but BLE only needs the permission responses wired in on the android activity side of things for the most part.  You can look at https://shinylib.net/client/hosting/manual/ 

You'll need to build the dependency injection setup yourself.

**Reply by redradist** [reply]
Okay, gotcha.

Is there anything else is needed for make it working ?
For example, should I use `.UseShiny()` function ? Or it will start working automatically after applying Manual hosting model ?

**Reply by redradist** [reply]
Also how to inject `IBleManager bleManager;` in manual hosting model case ??

**Reply by aritchie** [reply]
This is the stuff you'll have to figure out.  I don't know what the hosting model looks like in avalonia.  For injecting the BLE manager, again... it depends if you've got those services/pages on the hosting model as well

**Reply by redradist** [reply]
@aritchie I use Autofac instead of Avalonia injection mechanism ...

**Reply by redradist** [reply]
Looks like I have figured out how to do it:

`AutofacBuilder.Builder?.Register(c => Host.GetService<IBleManager>()).As<IBleManager>();`

**Reply by redradist** [reply]
By some reason Host.GetService<IBleManager>() returns null instead of BleManager defined in Shiny for Android

**Reply by redradist** [reply]
@aritchie I use the following code:
```csharp
AutofacBuilder.Builder?.Register((c) =>
        {
            var bleManager = Host.GetService<IBleManager>();
            return bleManager;
        }).As<IBleManager>();
```
where `AutofacBuilder.Builder` is `ContainerBuilder` from Autofac
Turns out that `var bleManager = ...` is `null`

**Reply by aritchie** [reply]
@redradist shiny is based in Microsoft extensions di.  As I said, you're on your own for this.  The code is open.  You can look at the hosting model in the source code
